### DISCUSSION TITLE
How do I override all config files with a custom name

### DISCUSSION THREAD

**Discussion by guru1306** [node-config/node-config#767] [Q&A] [answered]
Hi All

Thanks for creating an efficient module to load the config files.

Application ships with default.json and production.json located in ./config folder. I need to override these values with /user/override.json file. How do I achieve this? I found NODE_CONFIG_DIR to put all config folders. How do I mention the file name as override.json as the top override file.

Thanks
Guru

**Comment by jdmarshall** [comment, marked as answer]
For dev environments I like to use a symlink but I suspect what you’re asking is how do I install an app and allow individual users to override a property? If you can’t think of another way to do this (eg, if it’s one or two values value NODE_CONFIG may suffice)

NODE_CONFIG_DIR can contain a comma separated list of locations. It’s not advertised terribly loudly in the docs because it’s not the first thing one should try, but there are tests that assert it as a feature, and it is not on the deprecation list. If I were you I would build this stapling  process into the bootstrapping code of your application, to set both the app’s config directory and the one in $HOME at the same time.
